addons/udes_stock/models/stock_quant.py

```diff
@@ -210,3 +210,13 @@ def _split(self, qty):
             }
         )
         return new_quant
+
+    @api.model
+    def get_available_quantity(self, product, locations):
+        """Get available quantity of product_id within locations."""
+        domain = [("product_id", "=", product.id), ("location_id", "child_of", locations.ids)]
+        quants = self.search(domain)
```

Comment:
Q: Do we have any prefetch nasties with quants? Could end up with a lot of quants being returned on some bigger systems.

Reply:
Spoke with Miquel, we're only using two non-relational fields from quant, so let's go with disabling prefetching for now.

i.e.:

``` quants = self.search(domain).with_context(prefetch_fields=False)
    quants.read(["quantity", "reserved_quantity"], load="_classic_write")
```

Reply:
Added, please review.